File: leetcode/ds/1408. String Matching in an Array.cpp
class Solution {
public:
    vector<string> stringMatching(vector<string>& words) {
        set <string> s;
        vector <string> ans;
        for(int i = 0; i < words.size(); i++) {
            for(int j = 0; j < words.size(); j++) {
                if(i == j) continue;
                if(words[i].find(words[j]) != string::npos) {
                    s.insert(words[j]);
                }
            }
        }
        
        for(auto c : s) {
            ans.push_back(c);
        }
        
        return ans;
    }
};

File: leetcode/brute_force/1408. String Matching in an Array.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        length = len(words)
        answer = set()
        for i in range(length):
            for j in range(length):
                if i == j:
                    continue
                if words[j].find(words[i]) != -1:
                    answer.add(words[i])
        return list(answer)
